node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -205,16 +240,98 @@ impl OrderingProvider {
 			self.included_candidates.remove(&finalized_candidate);
 		}
 	}
+
+	/// Returns ancestors of `head` in the descending order, stopping
+	/// either at the block present in cache or the latest finalized block.
+	///
+	/// Suited specifically for querying non-finalized chains, thus
+	/// doesn't rely on block numbers.
+	///
+	/// Both `head` and last are **not** included in the result.
+	async fn get_block_ancestors<Sender: SubsystemSender>(
```

Comment:
Is `&mut self` needed, from the body it seems `&self` should suffice?

The function could avoid depending on `&mut self entirely, by passing two more variables and hence ease writing testing for this function without the need for much outside code.

Reply:
Calling `LruCache::get` requires mutable reference

Reply:
I see, then disregard. Good as is.